File: android/app/src/main/java/gg/strims/ppspp/rpc/VideoClient.kt
package gg.strims.ppspp.rpc
import gg.strims.ppspp.proto.*

class VideoClient(filepath: String) : RPCClient(filepath) {

    suspend fun openClient(
        arg: OpenVideoClientRequest = OpenVideoClientRequest()
    ): RPCResponseStream<VideoClientEvent> =
        this.callStreaming("Video/OpenClient", arg)

    suspend fun openServer(
        arg: OpenVideoServerRequest = OpenVideoServerRequest()
    ): VideoServerOpenResponse =
        this.callUnary("Video/OpenServer", arg)

    suspend fun writeToServer(
        arg: WriteToVideoServerRequest = WriteToVideoServerRequest()
    ): WriteToVideoServerResponse =
        this.callUnary("Video/WriteToServer", arg)

    suspend fun publishSwarm(
        arg: PublishSwarmRequest = PublishSwarmRequest()
    ): PublishSwarmResponse =
        this.callUnary("Video/PublishSwarm", arg)

    suspend fun startRTMPIngress(
        arg: StartRTMPIngressRequest = StartRTMPIngressRequest()
    ): StartRTMPIngressResponse =
        this.callUnary("Video/StartRTMPIngress", arg)

    suspend fun startHLSEgress(
        arg: StartHLSEgressRequest = StartHLSEgressRequest()
    ): StartHLSEgressResponse =
        this.callUnary("Video/StartHLSEgress", arg)

    suspend fun stopHLSEgress(
        arg: StopHLSEgressRequest = StopHLSEgressRequest()
    ): StopHLSEgressResponse =
        this.callUnary("Video/StopHLSEgress", arg)

}

File: src/lib/api/videoClient.ts
import * as pb from "../pb";
import { RPCHost } from "../rpc/host";
import { Readable as GenericReadable } from "../rpc/stream";

export default class Video {
  constructor(private readonly host: RPCHost) {}

  public openClient(
    arg: pb.IOpenVideoClientRequest = new pb.OpenVideoClientRequest()
  ): GenericReadable<pb.VideoClientEvent> {
    return this.host.expectMany(
      this.host.call("Video/OpenClient", new pb.OpenVideoClientRequest(arg))
    );
  }
  public openServer(
    arg: pb.IOpenVideoServerRequest = new pb.OpenVideoServerRequest()
  ): Promise<pb.VideoServerOpenResponse> {
    return this.host.expectOne(
      this.host.call("Video/OpenServer", new pb.OpenVideoServerRequest(arg))
    );
  }
  public writeToServer(
    arg: pb.IWriteToVideoServerRequest = new pb.WriteToVideoServerRequest()
  ): Promise<pb.WriteToVideoServerResponse> {
    return this.host.expectOne(
      this.host.call("Video/WriteToServer", new pb.WriteToVideoServerRequest(arg))
    );
  }
  public publishSwarm(
    arg: pb.IPublishSwarmRequest = new pb.PublishSwarmRequest()
  ): Promise<pb.PublishSwarmResponse> {
    return this.host.expectOne(
      this.host.call("Video/PublishSwarm", new pb.PublishSwarmRequest(arg))
    );
  }
  public startRTMPIngress(
    arg: pb.IStartRTMPIngressRequest = new pb.StartRTMPIngressRequest()
  ): Promise<pb.StartRTMPIngressResponse> {
    return this.host.expectOne(
      this.host.call("Video/StartRTMPIngress", new pb.StartRTMPIngressRequest(arg))
    );
  }
  public startHLSEgress(
    arg: pb.IStartHLSEgressRequest = new pb.StartHLSEgressRequest()
  ): Promise<pb.StartHLSEgressResponse> {
    return this.host.expectOne(
      this.host.call("Video/StartHLSEgress", new pb.StartHLSEgressRequest(arg))
    );
  }
  public stopHLSEgress(
    arg: pb.IStopHLSEgressRequest = new pb.StopHLSEgressRequest()
  ): Promise<pb.StopHLSEgressResponse> {
    return this.host.expectOne(
      this.host.call("Video/StopHLSEgress", new pb.StopHLSEgressRequest(arg))
    );
  }
}
